### PR TITLE
[FLINK-17996][table-planner-blink] NPE in CatalogTableStatisticsConverter.convertToColumnStats method

### PR DESCRIPTION
**Pull request by SteNicholas**
## What is the purpose of the change

*Hive catalog only supports a few kinds of column statistics, otherwise return null through `HiveStatsUtil#createTableColumnStats`. NullPointerException would occur in `CatalogTableStatisticsConverter#convertToColumnStatsMap` method which iterators the `CatalogColumnStatisticsDataBase` map.*

## Brief change log

  - *Method `CatalogTableStatisticsConverter#convertToColumnStatsMap` check `CatalogColumnStatisticsDataBase` whether is null. If `CatalogColumnStatisticsDataBase` is null, `CatalogColumnStatisticsDataBase` doesn't call `convertToColumnStats` method to convert to column stats.*

## Verifying this change

  - *Add `CatalogTableStatisticsConverterTest#testConvertToColumnStatsMapWithNullColumnStatisticsData` method to verify the null `CatalogColumnStatisticsDataBase` in column statistics data map scenario.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not** applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by SteNicholas** [comment]
@flinkbot run azure

File: flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/utils/CatalogTableStatisticsConverter.java [hunk 1]
```diff
@@ -61,12 +61,14 @@ public static TableStats convertToTableStats(
 		return new TableStats(rowCount, columnStatsMap);
 	}
 
-	private static Map<String, ColumnStats> convertToColumnStatsMap(
+	public static Map<String, ColumnStats> convertToColumnStatsMap(
```
Comment (lirui-apache) [inline flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/utils/CatalogTableStatisticsConverter.java:64]: Add VisibleForTesting annotation.

**Comment by lirui-apache** [comment]
@godfreyhe You might also want to have a look

**Comment by SteNicholas** [comment]
@JingsongLi @wuchong Would you please help to merge?

**Pull request merged**